src/clients/repositoryinfoclient.ts

```diff
@@ -0,0 +1,132 @@
+                Logger.LogDebug(`Found a team project collection for url '${serverUrl}' and collection name '${collectionName}'. ${collection.id}`);
+            } else {
+                //TODO: On-prem TFS
+            }
+
+            // Now get the team project information from the server
+            //TODO: This can throw
+            let project: TeamProject = await this.getProjectFromServer(coreApiClient, serverUrl, teamProjectName);
+            Logger.LogDebug(`Found a team project for url '${serverUrl}' and collection name '${collectionName}'. ${project.id}`);
+
+            //Now, create the JSON blob to send to new RepositoryInfo(repoInfo);
+            repoInfo = this.getTfvcRepoInfoBlob(serverUrl, collection.id, collection.name, collection.url, project.id, project.name, project.description, project.url);
+            repositoryInfo = new RepositoryInfo(repoInfo);
+            return repositoryInfo;
+        }
+        return repositoryInfo;
+    }
+
+    //TODO: serverUrl could be a guess (need to add accountName as collection?)
+    //RepositoryInfo uses repository.remoteUrl to set up accountUrl
+    private getTfvcRepoInfoBlob(serverUrl: string, collectionId: string, collectionName: string, collectionUrl: string,
+                                projectId: string, projectName: string, projectDesc: string, projectUrl: string): any {
+        //TODO: Not sure if remoteUrl should always be the same as serverUrl...?
+        return {
+            serverUrl: serverUrl,
+            collection: {
+                id: collectionId,
+                name: collectionName,
+                url: collectionUrl
+            },
+            repository: {
+                id: "00000000-0000-0000-0000-000000000000",
+                name: "NoNameTfvcRepository",
+                url: "http://xplatalm.visualstudio.com/repository-url-does-not-exist",
+                project: {
+                    id: projectId,
+                    name: projectName,
+                    description: projectDesc,
+                    url: projectUrl,
+                    state: 1,
+                    revision: 15
+                },
+                remoteUrl: serverUrl
+            }
+        };
+    }
+
+    private async getProjectFromServer(coreApiClient: CoreApiClient, remoteUrl: string, teamProjectName: string): Promise<TeamProject> {
+        return coreApiClient.GetTeamProject(remoteUrl, teamProjectName);
+    }
+
+    private async validateTfvcCollectionUrl(serverUrl: string, handler: VsoBaseInterfaces.IRequestHandler): Promise<boolean> {
+        try {
+            let repositoryClient: TeamServicesApi = new TeamServicesApi(serverUrl, [this._handler]);
+            await repositoryClient.validateTfvcCollectionUrl();
+            return true;
+        } catch (err) {
+            if (err.errorCode === "404") { //check this!!
+                return false;
+            } else {
+                throw err; //check this!!
```

Comment:
First, I should have added a TODO to 'check this!!' which flags it for me to come back and run through this code in an e2e scenario.  I'll do that (I typically ensure I have no TODOs left when I complete a User Story).  For reference, [here's the code I was referencing](https://github.com/Microsoft/vso-intellij/blob/master/plugin/src/com/microsoft/alm/plugin/context/ServerContextManager.java#L804-L813).